packages/annotations/src/vote.ts

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import {
+  ArcGISRequestError,
+  ArcGISAuthError,
+  IRequestOptions
+} from "@esri/arcgis-rest-request";
+
+import {
+  addFeatures,
+  deleteFeatures,
+  queryFeatures,
+  IAddFeaturesRequestOptions,
+  IDeleteFeaturesRequestOptions,
+  IAddFeaturesResult,
+  IDeleteFeaturesResult,
+  IQueryFeaturesResponse
+} from "@esri/arcgis-rest-feature-service";
+
+import { IAnnoFeature } from "./add";
+import { UserSession } from "@esri/arcgis-rest-auth";
+
+export interface IVoteRequestOptions extends IRequestOptions {
+  url: string;
+  annotation: IAnnoFeature;
+  /**
+   * Set to 'true' in order to vote another annotation down.
+   */
+  downVote?: boolean;
+  authentication: UserSession;
+}
+
+/**
+ * ```js
+ * import { vote } from "@esri/hub-annotations";
+ * //
+ * voteOnAnnotation({
+ *   url,
+ *   annotation: { attributes: {} },
+ *   authentication
+ * })
+ *   .then(response);
+ * ```
+ * Upvote (or downvote) another annotation.
+ * @param requestOptions - request options that may include authentication
+ * @returns A Promise that will resolve with the response from the service after attempting to vote on an annotation.
+ */
+export function voteOnAnnotation(
+  requestOptions: IVoteRequestOptions
+): Promise<IAddFeaturesResult | IDeleteFeaturesResult> {
+  const annotation = requestOptions.annotation;
+  const auth = requestOptions.authentication as UserSession;
+  const url = requestOptions.url;
+
+  if (!auth) {
+    return new Promise((resolve, reject) => {
+      reject(
+        new ArcGISAuthError(`Voting by anonymous users is not supported.`)
+      );
+    });
+  }
+
+  if (annotation.attributes.author === auth.username) {
+    return new Promise((resolve, reject) => {
+      reject(
+        new ArcGISRequestError(`Users may not vote on their own comment/idea.`)
+      );
+    });
+  }
+
+  // searchAnnotations() would make more xhrs for user metadata
+  return queryFeatures({
```

Comment:
i realized during the most recent refactor that `searchAnnotations()` creates even more network chatter (via appending user metadata), so i switched to using `queryFeatures()` instead.